linetools/analysis/abskin.py

```diff
@@ -0,0 +1,101 @@
+""" Utlities for kinematic analysis of absorption lines
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+import pdb
+
+from astropy import units as u
+from astropy.convolution import convolve, Box1DKernel
+
+
+def generate_stau(velo, flux, sig, kbin=22.*u.km/u.s, debug=False):
+    """ Generate the smoothed tau array for kinematic tests
+
+    Parameters
+    ----------
+    velo : Quantity array (usually km/s)
+    flux : Quantity array (flux)
+    sig :  Quantity array (sig)
+    kbin : Quantity (velocity), optional
+      Kernel size for Gaussian smoothing of optical depth array
+    debug : bool, optional
+
+    Returns
+    -------
+    stau : array
+       Smoothed tau array
+    """
+    # Velocity array
+    npix = len(velo)
+    pix = np.arange(npix).astype(int)
+
+    # Calculate dv
+    dv = np.abs(np.median(velo-np.roll(velo,1)))
+
+    # Test for bad pixels
+    badzero = np.where((flux == 0) & (sig <= 0))[0]
```

Comment:
Should this be `(flux == 0) | (sig <= 0)`? Also maybe change to ~(sig > 0) so it will catch NaN values.